erpnext/utilities/bulk_transaction.py

```diff
@@ -0,0 +1,138 @@
+import json
+from datetime import date, datetime
+
+import frappe
+
+
+@frappe.whitelist()
+def transaction_processing(data, from_doctype, to_doctype):
+	if isinstance(data, str):
+		deserialized_data = json.loads(data)
+
+	else:
+		deserialized_data = data
+
+	length_of_data = len(deserialized_data)
+
+	if length_of_data > 10:
+		frappe.msgprint("Started a background job to create {1} {0}".format(to_doctype,length_of_data))
+		frappe.enqueue(job, deserialized_data=deserialized_data, from_doctype=from_doctype, to_doctype=to_doctype)
+	else:
+		job(deserialized_data, from_doctype, to_doctype)
+
+def job(deserialized_data, from_doctype, to_doctype):
+	failed_history = []
+	i = 0
+	for d in deserialized_data:
+		failed = []
+
+		try:
+			i+=1
+			doc_name = d.get('name')
+			task(doc_name, from_doctype, to_doctype)
+
+		except Exception as e:
```

Comment:
This has potential commit failed data. 

E.g. if exception is thrown in `on_submit` method, a lot of data is already written, docstatus already changed etc etc that has to be rolledback in case of exception. 

Either use commit + rollback or new feature of savepoints: https://github.com/frappe/frappe/pull/15379

Reply:
Did the changes